### PR TITLE
Add Registration Sample App for Kotlin

### PR DESCRIPTION
**Pull request by dzou**
This adds a Registration sample application for Kotlin. This sample app exercises the pub/sub and cloud SQL integrations from spring gcp.

In reference to #1299.

### CODE REVIEW

File: spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/README.adoc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+2. Create a Google Cloud MySQL instance and a database within the instance.
+
+- You may create a Google Cloud MySQL instance on the https://console.cloud.google.com/sql/instances[Google Cloud Console Cloud SQL page].
+- In the same page, you may create a database within your Cloud SQL instance. Name your database `registrants`.
+
+3. Create a Pub/Sub topic called `registrations` and a subscription for the topic called `registrations-sub`.
+
+- Go to the https://console.cloud.google.com/cloudpubsub/topicList[Google Cloud Console Pub/Sub topics page]
+and create a topic called `registrations`.
```
Comment (artembilan) [inline spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/README.adoc]: Please, concatenate this with the previous line.

File: spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/src/main/kotlin/com/example/app/Controller.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+			@RequestParam("firstName") firstName: String,
+			@RequestParam("lastName") lastName: String,
+			@RequestParam("email") email: String): RedirectView {
+
+		pubSubTemplate.publish(REGISTRATION_TOPIC, "$firstName:$lastName:$email")
+		return RedirectView("/")
+	}
+
+	@GetMapping("/registrants")
+	fun registerPerson(): ModelAndView {
```
Comment (artembilan) [inline spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/src/main/kotlin/com/example/app/Controller.kt]: I don't think the name of this method reflects a reality.
More over it is clashes with the previous one.

File: spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/README.adoc [hunk 3]
```diff
@@ -0,0 +1,48 @@
+= Spring Cloud GCP Kotlin Sample App
+
+This is a simple Kotlin Spring Boot application which demonstrates how to integrate with Spring Cloud GCP libraries from Kotlin.
+It sets up a registration service in which a someone publishes registrant names to a Cloud Pub/Sub topic; these names are then persisted to a Cloud SQL database.
+
+This application demonstrates the following Spring Cloud GCP integrations:
```
Comment (meltsufin) [inline spring-cloud-gcp-kotlin-samples/spring-cloud-gcp-kotlin-app-sample/README.adoc]: You should also mention which parts of Spring it demonstrates, namely, Spring Integrations and Spring Data JPA.

**Comment by dzou** [comment]
Done. I've created #1323 to track making this Kotlin app more discoverable. I can make the refdoc update and we can also reference our app from the kotlin team's samples directory.

**Pull request merged**